### PR TITLE
Document custom CPython roles.

### LINKED ISSUE #1091: Document `issue`, `gh-issue` and `source` roles
**Describe the enhancement or feature you'd like**

We define three Sphinx roles in `pyspecific.py`: `issue`, `gh-issue` and `source`.

https://github.com/python/cpython/blob/a679c3d58d10aafd9ac9355fdd16151607e37d65/Doc/tools/extensions/pyspecific.py#L54-L92

* `issue` is for issues on the old https://bugs.python.org, and so won't be used much anymore, but still exists in lots of old docs
* `gh-issue` is for new issues on GitHub
* `source` is for linking to source files on GitHub

Let's document them somewhere on https://devguide.python.org/documentation/markup/#roles

**Describe alternatives you've considered**

Don't document them = status quo.

Remove them = make the docs worse.

### PR DESCRIPTION
**Pull request by ezio-melotti**
Fixes #1091.

----
:books: Documentation preview :books:: https://cpython-devguide--1092.org.readthedocs.build/

### CODE REVIEW

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by encukou** [comment]
Is it ``` :bpo:`ID` ``` or ``` :issue:`ID` ```?

File: documentation/markup.rst [hunk 1]
```diff
@@ -30,6 +30,8 @@ external links          ```Link text <https://example.com>`_``      :ref:`hyperl
 roles w/ custom text    ``:role:`custom text <target>```            :ref:`roles`
 roles w/ only last part ``:role:`~hidden.hidden.visible```          :ref:`roles`
 roles w/o link          ``:role:`!target```                         :ref:`roles`
+issues                  ``:gh:`ID```, ``:bpo:`ID```                 :ref:`roles`
```
Comment (hugovk) [inline documentation/markup.rst:33]:
```suggestion
issues                  ``:gh:`ID```, ``:issue:`ID```                 :ref:`roles`
```

File: documentation/markup.rst [hunk 2]
```diff
@@ -30,6 +30,8 @@ external links          ```Link text <https://example.com>`_``      :ref:`hyperl
 roles w/ custom text    ``:role:`custom text <target>```            :ref:`roles`
 roles w/ only last part ``:role:`~hidden.hidden.visible```          :ref:`roles`
 roles w/o link          ``:role:`!target```                         :ref:`roles`
+issues                  ``:gh:`ID```, ``:issue:`ID```                 :ref:`roles`
```
Comment (ezio-melotti) [inline documentation/markup.rst:33]:
```suggestion
issues                  ``:gh:`ID```, ``:issue:`ID```               :ref:`roles`
```

**Pull request merged**